server/controllers/socialLoginController.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable object-curly-newline */
+import Model from '../models';
+import Authenticator from '../middlewares/authenticator';
+
+
+const { User } = Model;
+const { generateToken } = Authenticator;
+
+/**
+ * @class SocialLoginController
+ */
+class SocialLoginController {
+  /**
+   * @description - find  or create a new user and pass the user to the request body
+   * @param {object} user
+   * @param {function} done
+   * @returns {object} createOrFindUser
+   */
+  static modelQuery(user, done) {
+    User.findOrCreate({
+      where: { email: user.email },
+      // users signing up via socialmedia should be verified  authomatically
+      // given a random password since password cannot be null
+      defaults: { isVerified: true, username: user.email, password: '5kb3455uf884545i44i' },
+    }).spread((foundOrCreated, created) => {
```

Comment:
what exactly was found here @udochukwu, please use a descriptive name for the variable here too